app/segments/drill/[id]/submission/input.js

```diff
@@ -137,6 +141,136 @@ async function uploadAttempt(
 }
 
 //TODO: Create a function to check leaderboard and update if needed
+function checkLeaderboardUpdate(uploadData, drillInfo, currentLeaderboard) {
+  const leaderboardData = currentLeaderboard["data"];
+
+  //used if an attempt already exists
+  const currentBest =
+    leaderboardData[uploadData.uid]?.[drillInfo.mainOutputAttempt]?.value;
+
+  const newAttempt = {
```

Comment:
Move newAttempt into uploadNewLeaderboard since this structure is not specifically needed in this function. You can thenjust pass uploadData into uploadNewLeaderboard instead of splitting it out like you're doing now

Reply:
Also either pass mainOutputAttempt or the entire drillInfo (prob the former)

Reply:
I think both of these were resolved; newAttempt is declared in uploadNewLeaderboard, and mainOutputAttempt is passed as input arg